unified-installer/astra-unified-installer.sh

```diff
@@ -1117,12 +1179,177 @@ is_astra_registry() {
     local -r registry_url="$1"
     if [ -z "$registry_url" ]; then fatal "no registry_url given"; fi
 
-    if echo "$registry" | grep -q "cr." && echo "$registry" | grep -q "astra"; then
+    if echo "$registry_url" | grep -q "cr." && echo "$registry_url" | grep -q "astra"; then
         return 0
     fi
     return 1
 }
 
+# find_astra_registry loops through the given list of registries and returns the first one which matches the
+# pattern of an Astra registry.
+find_astra_registry() {
```

Comment:
This function and the connector/trident-specific ones below are for determining if we want to create **astra-regcred** or not.